tls/src/client.rs

```diff
@@ -206,13 +235,87 @@ impl Client {
         let stream = match TcpStream::connect(addr).await {
             Ok(s) => s,
             Err(e) => {
-                println!("{:?}", e);
+                println!("{e:?}");
                 return Err(e.into());
             }
         };
 
-        let stream = connector.connect(dnsname, stream).await?;
-        // TODO: Measurement Attestations
+        let mut stream = connector.connect(dnsname, stream).await?;
+
+        // send Nonce to server
+        let nonce = Nonce::from_platform_rng()?;
+        send_msg(&mut stream, nonce.as_ref()).await?;
+
+        // get Nonce from server
+        let server_nonce = recv_msg(&mut stream).await?;
+        let server_nonce = Nonce::try_from(server_nonce)?;
+
+        let (cert_chain, log, attestation) =
+            crate::keys::get_attest_data(&attest_config, &server_nonce)?;
+
+        dice_verifier::verify_attestation(
+            &cert_chain[0],
+            &attestation,
+            &log,
+            &server_nonce,
+        )?;
+
+        // send client attestation cert chain to server
+        let cert_chain_der = certs_to_der(&cert_chain)?;
+        send_msg(&mut stream, &cert_chain_der).await?;
+
+        // get & verify server attestation cert chain
+        let server_cert_chain = recv_msg(&mut stream).await?;
+        let server_cert_chain = certs_from_der(&server_cert_chain)?;
+        let root =
+            dice_verifier::verify_cert_chain(&server_cert_chain, Some(&roots))?;
+        let client_platform_id =
+            dice_mfg_msgs::PlatformId::try_from(&server_cert_chain)?;
```

Comment:
this name is misleading: we're operating on the server's cert chain so the platform id is for the server, not client